source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -1940,8 +1940,28 @@ unsigned SwiftExpressionParser::Parse(DiagnosticManager &diagnostic_manager,
     log->PutCString(s.c_str());
   }
 
-  runSILDiagnosticPasses(*sil_module);
+  // Serialize the file if modules directory is set.
+  if (auto expr_module_dir = swift_ast_ctx->GetReplExprModulesDir()) {
```

Comment:
Could this, and the chunk of code on L2043 that initializes `module` happen at the same time, to keep it all in the same place? It would serialize it to a file and then immediately load it again.

Also, would it be possible to put it all around the end of `ParseAndImport` so that `ParseAndImport` returns a `module` that has been all correctly processed and no one gets confused that `parsed_expr->module` is not the "right" module to use in some places any more?

Reply:
Unfortunately, no. The module needs to be serialized just after SILGen. I added a brief explanation as to why that is necessary. See above.